### PR TITLE
Documenting the config system

### PR DESCRIPTION
**Pull request by leotrs**
This PR adds a first pass at documenting the configuration system, including the CLI flags available, and the cascading config file system. See the file `docs/source/configuration.rst`. In particular, the whole output of `manim -h` is copy&pasted into this document, as a quick&dirty way of documenting the available flags. I don't love this, but it saved a ton of time when I was writing this document. A good reason to keep it this way is that it is included within a `doctest` directive. This means that running `make doctest` from the `docs/` folder will tell us if the documentation is out of date. Thus it will be easy to always have it mirror the true, up-to-date output of `manim -h`.

This PR also includes general small changes to the description of CLI flags, as seen in `manim -h`.

This PR also changes the `--color` flag to `--background_color` so that it matches the configuration option used in the config files `background_color = BLACK`.

Finally, it also includes some changes made by `black`.

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -1,4 +1,286 @@
...
+create a special config file that will apply to all scenes rendered by that
+user. This is referred to as the **user-wide** config file, and it will apply
+regardless of where manim is executed from, and regardless of where the scene
+script is stored.
+
+The user-wide config file lives in a special folder, depending on the operating
+system.
+
+* Windows: :code:`UserDirectory`/AppData/Roaming/Manim/manim.cfg
+* Mac OSX: :code:`UserDirectory`/config/manim/manim.cfg
```
Comment (PgBiel) [inline docs/source/configuration.rst:219]:
```suggestion
* MacOS: :code:`UserDirectory`/config/manim/manim.cfg
```

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -1,4 +1,286 @@
...
+
+   manim <file.py> SceneName -o myscene -c RED
+
+To summarize, the order of precedence for configuration options is: *user-wide
+config file < folder-wide config file < CLI flags*.
+
+
+.. note:: There is also a **library-wide** config file that determines manim's
+	  default behavior, and applies to every user of the library.  It has
+	  the least precedence, and the **user should not** try to modify it.
```
Comment (PgBiel) [inline docs/source/configuration.rst:284]: nitpick, idk looks better
```suggestion
	  the least precedence, and **the user should not try to modify it**.
```

File: manim/utils/config_utils.py [hunk 3]
```diff
@@ -240,14 +240,14 @@ def _parse_cli(arg_list, input=True):
...
-        help="Render the scene as a movie file",
+        help="Render the scene as a movie file (this is on by default)",
     )
     parser.add_argument(
         "-s",
         "--save_last_frame",
         action="store_const",
         const=True,
-        help="Save the last frame (and do not save movie)",
+        help="Save the last frame only (no movie file generated)",
```
Comment (PgBiel) [inline manim/utils/config_utils.py:250]: nitpick
```suggestion
        help="Save the last frame only (no movie file is generated)",
```

**Comment by leotrs** [comment]
@PgBiel done!

**Pull request merged**